pkg/gadget-context/gadget-context.go

```diff
@@ -412,6 +399,35 @@ func (c *GadgetContext) LoadGadgetInfo(info *api.GadgetInfo, paramValues api.Par
 	return nil
 }
 
+func (c *GadgetContext) LoadGadgetInfoAndRun(info *api.GadgetInfo, paramValues api.ParamValues, remoteDone chan any) error {
+	err := c.LoadGadgetInfo(info, paramValues, nil)
+	if err != nil {
+		return err
+	}
+
+	if err := c.start(c.localOperators); err != nil {
+		return fmt.Errorf("starting local operators: %w", err)
+	}
+
+	c.Logger().Debugf("running...")
+
+	go func() {
+		// TODO: Client shouldn't need to wait for the timeout. It should be
+		// managed only on the server side.
+		WaitForTimeoutOrDone(c)
+
+		select {
+		case <-remoteDone:
+		// TODO Do we need a timeout here? I think not, since a second Ctrl+C does an exit
+		case <-time.After(3 * time.Second):
```

Comment:
I think 3 seconds is too short. IMO it should be > 10 seconds so the server has time to send the information to the client.

Reply:
See https://github.com/inspektor-gadget/inspektor-gadget/pull/4396#discussion_r2100732546